omegaconf/nodes.py

```diff
@@ -400,6 +401,62 @@ def __deepcopy__(self, memo: Dict[int, Any]) -> "EnumNode":
         return res
 
 
+class LiteralNode(ValueNode):
+    def __init__(
+        self,
+        literal_type: Any,  # cannot Type[Literal] because Literal requires an argument
+        value: Optional[Union[Enum, str, int, bool]] = None,
+        key: Any = None,
+        parent: Optional[Container] = None,
+        is_optional: bool = True,
+        flags: Optional[Dict[str, bool]] = None,
+    ):
+        if not is_literal_annotation(literal_type):
+            raise ValidationError(
+                f"LiteralNode can only operate on Literal annotation ({literal_type})"
+            )
+        self.literal_type = literal_type
+        if hasattr(self.literal_type, "__args__"):
+            self.fields = list(self.literal_type.__args__)  # pragma: no cover
+        elif hasattr(self.literal_type, "__values__"):  # pragma: no cover
+            self.fields = list(self.literal_type.__values__)  # pragma: no cover
```

Comment:
Maybe make these branches explicitly correspond to python versions. So I think
```suggestion
        if sys.version_info >= (3, 7):
            self.fields = list(self.literal_type.__args__)  # pragma: no cover
        else:
            self.fields = list(self.literal_type.__values__)  # pragma: no cover
```
but I'm not really sure about this.

Reply:
Why do you feel this is better? Philosophically I tend to think that directly checking is the right way to do things, in case e.g. the attribute name every switches back or something equally cursed. This also has the benefit that mypy/pylance/other syntax parsers can correctly infer that the attribute exists, which is left implicit if you're going by version.

If this is about readability, I'd be happy to add comments saying why the check is necessary.

Reply:
Right, I was just thinking that at some point in the future, omegaconf might want to drop support for Python 3.6 and then it would be useful that you can directly see that this if-else check has become unnecessary.

But, yes, a comment is likely already sufficient.